libcst/codegen/gen_visitor_functions.py

```diff
@@ -0,0 +1,181 @@
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+# pyre-strict
+import inspect
+from collections import defaultdict
+from dataclasses import fields
+from typing import Dict, Generator, List, Mapping, Sequence, Set, Type
+
+import libcst as cst
+
+
+DO_NOT_CARE_TYPES: Set[str] = {"CSTNode", "ABC", "object"}
+
+
+def _get_nodes() -> Generator[Type[cst.CSTNode], None, None]:
+    """
+    Grab all CSTNodes that are not a superclass. Basically, anything that a
+    person might use to generate a tree.
+    """
+
+    # pyre-ignore __dir__ exists on all objects.
+    for name in cst.__dir__():
+        if name.startswith("__") and name.endswith("__"):
+            continue
+        if name == "CSTNode":
+            continue
+
+        node = getattr(cst, name)
+        try:
+            if issubclass(node, cst.CSTNode):
+                yield node
+        except TypeError:
+            # This isn't a class, so we don't care about it.
+            pass
+
+
+nodes: Sequence[Type[cst.CSTNode]] = list(_get_nodes())
+classmap: Dict[Type[cst.CSTNode], Set[Type[cst.CSTNode]]] = {}
+for node in nodes:
+    # Map the base classes for this node
+    classmap[node] = set(inspect.getmro(node))
+
+
+def _get_bases(node: Type[cst.CSTNode]) -> Generator[Type[cst.CSTNode], None, None]:
+    for base in classmap[node]:
+        if base.__name__ in DO_NOT_CARE_TYPES:
+            # Ignore bases that are Python implementation details.
+            continue
+        if base not in classmap:
+            # Ignore bases that aren't in our tracking list.
+            continue
+        if base.__name__.startswith("_Base"):
```

Comment:
These underscore-prefixed members shouldn't be exported from the top-level libcst package, and therefore shouldn't be in `classmap` anyways.

Reply:
They show up because we ask for all of the bases using inspect.getmro().

Reply:
Yes, but then you filter things out with `if base not in classmap`. No class beginning with `_Base` should be in `classmap`'s set of keys.